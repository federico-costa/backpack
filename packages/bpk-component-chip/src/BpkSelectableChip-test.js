/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2021 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import React from 'react';
import renderer from 'react-test-renderer';

import BpkSelectableChip from './BpkSelectableChip';
import { CHIP_TYPES } from './commonTypes';

// Just a convenience wrapper that includes the default props so we don't
// have to keep writing them for each test.
const TestChip = ({ ...rest }) => (
  <BpkSelectableChip onClick={() => null} accessibilityLabel="Toggle" {...rest}>
    Toggle me
  </BpkSelectableChip>
);

describe('BpkSelectableChip', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<TestChip />);
    expect(tree).toMatchSnapshot();
  });

  Object.keys(CHIP_TYPES).forEach(chipType => {
    it(`should render correctly with type="${chipType}"`, () => {
      const tree = renderer.create(<TestChip type={chipType} />);
      expect(tree).toMatchSnapshot();
    });

    it(`should render correctly with type="${chipType}" and selected`, () => {
      const tree = renderer.create(<TestChip type={chipType} selected />);
      expect(tree).toMatchSnapshot();
    });

    it(`should render correctly with type="${chipType}" and disabled`, () => {
      const tree = renderer.create(<TestChip type={chipType} disabled />);
      expect(tree).toMatchSnapshot();
    });

    it(`should render correctly with type="${chipType}", selected and disabled`, () => {
      const tree = renderer.create(
        <TestChip type={chipType} selected disabled />,
      );
      expect(tree).toMatchSnapshot();
    });

    it(`should render correctly with type="${chipType}" and a leading accessory view`, () => {
      const tree = renderer.create(
        <TestChip
          type={chipType}
          leadingAccessoryView={<span>Leading</span>}
        />,
      );
      expect(tree).toMatchSnapshot();
    });

    it(`should render correctly with type="${chipType}" and a trailing accessory view`, () => {
      const tree = renderer.create(
        <TestChip
          type={chipType}
          trailingAccessoryView={<span>Trailing</span>}
        />,
      );
      expect(tree).toMatchSnapshot();
    });

    it(`should render correctly with type="${chipType}" and both a leading and trailing accessory view`, () => {
      const tree = renderer.create(
        <TestChip
          type={chipType}
          leadingAccessoryView={<span>Leading</span>}
          trailingAccessoryView={<span>Trailing</span>}
        />,
      );
      expect(tree).toMatchSnapshot();
    });
  });

  it('should render correctly with a "className" attribute', () => {
    const tree = renderer.create(<TestChip className="custom-class" />);
    expect(tree).toMatchSnapshot();
  });
});
