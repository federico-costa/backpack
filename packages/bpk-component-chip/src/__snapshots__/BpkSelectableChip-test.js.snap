// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkSelectableChip should render correctly 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--primary"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with a "className" attribute 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--primary custom-class"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="light" 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--light"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="light" and a leading accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--light"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  <span
    className="bpk-chip__leading-accessory-view"
  >
    <span>
      Leading
    </span>
  </span>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="light" and a trailing accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--light"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
  <span
    className="bpk-chip__trailing-accessory-view"
  >
    <span>
      Trailing
    </span>
  </span>
</button>
`;

exports[`BpkSelectableChip should render correctly with type="light" and both a leading and trailing accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--light"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  <span
    className="bpk-chip__leading-accessory-view"
  >
    <span>
      Leading
    </span>
  </span>
  Toggle me
  <span
    className="bpk-chip__trailing-accessory-view"
  >
    <span>
      Trailing
    </span>
  </span>
</button>
`;

exports[`BpkSelectableChip should render correctly with type="light" and disabled 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--light bpk-chip--disabled"
  disabled={true}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="light" and selected 1`] = `
<button
  aria-checked={true}
  className="bpk-chip bpk-chip--light bpk-chip--light-selected"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="light", selected and disabled 1`] = `
<button
  aria-checked={true}
  className="bpk-chip bpk-chip--light bpk-chip--disabled"
  disabled={true}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="primary" 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--primary"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="primary" and a leading accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--primary"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  <span
    className="bpk-chip__leading-accessory-view"
  >
    <span>
      Leading
    </span>
  </span>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="primary" and a trailing accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--primary"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
  <span
    className="bpk-chip__trailing-accessory-view"
  >
    <span>
      Trailing
    </span>
  </span>
</button>
`;

exports[`BpkSelectableChip should render correctly with type="primary" and both a leading and trailing accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--primary"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  <span
    className="bpk-chip__leading-accessory-view"
  >
    <span>
      Leading
    </span>
  </span>
  Toggle me
  <span
    className="bpk-chip__trailing-accessory-view"
  >
    <span>
      Trailing
    </span>
  </span>
</button>
`;

exports[`BpkSelectableChip should render correctly with type="primary" and disabled 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--primary bpk-chip--disabled"
  disabled={true}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="primary" and selected 1`] = `
<button
  aria-checked={true}
  className="bpk-chip bpk-chip--primary bpk-chip--primary-selected"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="primary", selected and disabled 1`] = `
<button
  aria-checked={true}
  className="bpk-chip bpk-chip--primary bpk-chip--disabled"
  disabled={true}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="success" 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--success"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="success" and a leading accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--success"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  <span
    className="bpk-chip__leading-accessory-view"
  >
    <span>
      Leading
    </span>
  </span>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="success" and a trailing accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--success"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
  <span
    className="bpk-chip__trailing-accessory-view"
  >
    <span>
      Trailing
    </span>
  </span>
</button>
`;

exports[`BpkSelectableChip should render correctly with type="success" and both a leading and trailing accessory view 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--success"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  <span
    className="bpk-chip__leading-accessory-view"
  >
    <span>
      Leading
    </span>
  </span>
  Toggle me
  <span
    className="bpk-chip__trailing-accessory-view"
  >
    <span>
      Trailing
    </span>
  </span>
</button>
`;

exports[`BpkSelectableChip should render correctly with type="success" and disabled 1`] = `
<button
  aria-checked={false}
  className="bpk-chip bpk-chip--success bpk-chip--disabled"
  disabled={true}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="success" and selected 1`] = `
<button
  aria-checked={true}
  className="bpk-chip bpk-chip--success bpk-chip--success-selected"
  disabled={false}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;

exports[`BpkSelectableChip should render correctly with type="success", selected and disabled 1`] = `
<button
  aria-checked={true}
  className="bpk-chip bpk-chip--success bpk-chip--disabled"
  disabled={true}
  onClick={[Function]}
  role="checkbox"
  title="Toggle"
  type="button"
>
  Toggle me
</button>
`;
